code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1737,6 +1737,11 @@ func (sc *StorageSmartContract) finishAllocation(
 			return common.NewError("fini_alloc_failed",
 				"emitting blobber "+b.ID+": "+err.Error())
 		}
+		err = removeAllocationFromBlobber(sc, d.BlobberID, d.BlobberAllocationsPartitionLoc, alloc.ID, balances)
```

Comment:
We can do group removing by getting all blobbers that have the same partition location and remove them in one parititon read/write.

Reply:
oh yes, you right. They are in different partitions.